catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java

```diff
@@ -0,0 +1,384 @@
+
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_RED_LEFT, red));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_GREEN_LEFT, green));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_BLUE_LEFT, blue));
+	}
+
+	@Override
+	public void setRightRGBLightColor(int red, int green, int blue) {
+		red = checkRBGValue(red);
+		green = checkRBGValue(green);
+		blue = checkRBGValue(blue);
+
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_RED_RIGHT, red));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_GREEN_RIGHT, green));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_BLUE_RIGHT, blue));
+	}
+
+	private int percentToSpeed(int percent) {
+		if (percent <= 0) {
+			return 0;
+		}
+		if (percent >= 100) {
+			return 255;
+		}
+
+		return (int)(percent * 2.55);
+	}
+
+	private int checkRBGValue(int rgbValue) {
+		if (rgbValue > 255) {
+			return 255;
+		}
+
+		if (rgbValue < 0) {
+			return 0;
+		}
+
+		return rgbValue;
+	}
+
+	@Override
+	public String getName() {
+		return "Phiro";
+	}
+
+	@Override
+	public Class<? extends BluetoothDevice> getDeviceType() {
+		return PHIRO;
+	}
+
+	@Override
+	public void setConnection(BluetoothConnection connection) {
+		this.btConnection = connection;
+	}
+
+	@Override
+	public void disconnect() {
+
+		if (firmata == null) {
+			return;
+		}
+
+		try {
+			resetPins();
+			this.reportSensorData(false);
+			firmata.clearListeners();
+			firmata.getSerial().stop();
+			isInitialized = false;
+			firmata = null;
+		} catch (SerialException e) {
+			Log.d(TAG, "Error stop phiro pro serial");
+		}
+	}
+
+	@Override
+	public boolean isAlive() {
+		if (firmata == null) {
+			return false;
+		}
+
+		try {
+			firmata.send(new ReportFirmwareVersionMessage());
+			return true;
+		} catch (SerialException e) {
+			return false;
+		}
+	}
+
+	public void reportFirmwareVersion() {
+		if (firmata == null) {
+			return;
+		}
+
+		try {
+			firmata.send(new ReportFirmwareVersionMessage());
+		} catch (SerialException e) {
+			Log.d(TAG, "Firmata Serial error, cannot send message.");
+		}
+	}
+
+	@Override
+	public int getSensorValue(Sensors sensor) {
+		switch (sensor) {
+			case PHIRO_BOTTOM_LEFT:
+				return phiroListener.getBottomLeftSensor();
+			case PHIRO_BOTTOM_RIGHT:
+				return phiroListener.getBottomRightSensor();
+			case PHIRO_FRONT_LEFT:
+				return phiroListener.getFrontLeftSensor();
+			case PHIRO_FRONT_RIGHT:
+				return phiroListener.getFrontRightSensor();
+			case PHIRO_SIDE_LEFT:
+				return phiroListener.getSideLeftSensor();
+			case PHIRO_SIDE_RIGHT:
+				return phiroListener.getSideRightSensor();
+		}
+
+		return 0;
+	}
+
+	public int getSensorValue(int numberFromBrick) {
+		switch (numberFromBrick) {
+			case 0:
+				return phiroListener.getSideRightSensor();
+			case 1:
+				return phiroListener.getFrontRightSensor();
+			case 2:
+				return phiroListener.getBottomRightSensor();
+			case 3:
+				return phiroListener.getBottomLeftSensor();
+			case 4:
+				return phiroListener.getFrontLeftSensor();
+			case 5:
```

Comment:
Is there a mapping from 0..5 to the sensor or a place where this place is made explicit?